File: src/Module.TerritoryData.Library/Interfaces/TerritoryDataRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Module.TerritoryData.Library.Interfaces
{
    public interface TerritoryDataRepository
    {
        Models.Country GetCountry(string countryCode);
        Models.CountryList GetCountryList();
        Models.Zone1 GetZone1(string zone1Code);
        Models.Zone1List GetZone1List(Models.SearchParams searchParams);
        Models.Zone2 GetZone2(string zone2Code);
        Models.Zone2List GetZone2List(Models.SearchParams searchParams);
        Models.Zone3 GetZone3(string zone3Code);
        Models.Zone3List GetZone3List(Models.SearchParams searchParams);
        Models.City GetCity(string cityCode);
        Models.CityList GetCityList(Models.SearchParams searchParams);
        Models.Address Address(string addressCode);
        Models.AddressList GetAddressList(Models.SearchParams searchParams);

    }
}

File: java/module/territorydata/library/interfaces/TerritoryDataRepository.java
package module.territorydata.library.interfaces;

public interface TerritoryDataRepository
{
	models.Country GetCountry(String countryCode);
	models.CountryList GetCountryList();
	models.Zone1 GetZone1(String zone1Code);
	models.Zone1List GetZone1List(models.SearchParams searchParams);
	models.Zone2 GetZone2(String zone2Code);
	models.Zone2List GetZone2List(models.SearchParams searchParams);
	models.Zone3 GetZone3(String zone3Code);
	models.Zone3List GetZone3List(models.SearchParams searchParams);
	models.City GetCity(String cityCode);
	models.CityList GetCityList(models.SearchParams searchParams);
	models.Address Address(String addressCode);
	models.AddressList GetAddressList(models.SearchParams searchParams);

}